docs/developers/enabling-ha.md

```diff
@@ -0,0 +1,75 @@
+# HA support for Tekton Pipeline controllers
+
+- [Overview](#overview)
+- [Configuring HA](#configuring-ha)
+  - [Configuring the controller replicas](#configuring-the-controller-replicas)
+  - [Configuring the leader election](#configuring-the-leader-election)
+- [Disabling leader election](#disabling-leader-election)
+  - [Use the disable-ha flag](#use-the-disable-ha-flag)
+  - [Scale down your replicas](#scale-down-your-replicas)
+
+## Overview
+
+---
+This document is aimed at helping maintainers/developers when configuring High Availability(HA) support for the Tekton Pipeline [controller deployment](../../config/controller.yaml).
+
+HA support allows components to remain operational when a disruption occurs. This is achieved by following an active/passive model, where only one instance is active at a time.
+
+In the case of the Tekton pipeline controller, when the active controller fails or is deleted, another instance of that controller should become the active one, taking over all related API requests that the previous active controller was serving. The active instance is usually refer as the `leader`.
```

Comment:
Uhm, in my understanding resources are partitioned and each reconciler is leader of its own bucket of resource. In case of failure of a reconciler, its bucket is spread across the remaining buckets to keep the load even.  This approach provides both horizontal scaling as well as fault tolerance at the same time.
@mattmoor ^^^

Reply:
Yes,  you are correct, if this is the case I think these two paragraphs should be rewritten.

From

> HA support allows components to remain operational when a disruption occurs. This is achieved by following an active/passive model, where only one instance is active at a time.

To
> HA support allows components to remain operational when a disruption occurs. This is achieved by following an approach where reconciliation events are distributed in buckets, and buckets are mapped to certain replicas. This mean that all replicas of a component will be processing buckets of workload. In the case of a replica disruption, the remaining replicas will inherit those buckets.

but I think @mattmoor could provide a more accurate text.

Reply:
I updated the text with a more accurate description based on the behaviour of the active/active, fyi